frontend/src/components/ScoreContainer.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react';
+import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
+import 'react-circular-progressbar/dist/styles.css';
+import userService from '../services/userService';
+
+const ScoreContainer: React.FC = () => {
+    const [totalScore, setTotalScore] = useState<number | null>(null);
+    const [isLoading, setIsLoading] = useState(true);
+
+    const getColor = (score: number) => {
+        if (score >= 90) return '#2196F3';
+        if (score >= 75) return '#4CAF50';
+        if (score >= 50) return '#FFEB3B';
+        if (score >= 25) return '#FF9800';
+        return '#F44336';
+    };
+
+    useEffect(() => {
+        const fetchTotalScore = async () => {
+            setIsLoading(true);
+            try {
+                const score = await userService.getTotalScore();
+                setTotalScore(parseFloat(score));
```

Comment:
Here you are missing the check on isNaN that you did previously in the ThankYou component. Besides, I would return a number instead of a string from the BE, or is it just the wrong type assigned in Typescript and should it be a number, instead?

Reply:
Fixed this